src/components/Kanban/Kanban.tsx

```diff
@@ -0,0 +1,272 @@
+import "@Components/Kanban/kanban.css";
+import { useKanban, useToggleKanban } from "@Root/src/store";
+import { IconContext } from "react-icons";
+import { IoCloseSharp } from "react-icons/io5";
+import { BsPlus } from "react-icons/bs";
+import { RxPencil2, RxCheck } from "react-icons/rx";
+import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
+import { FormEvent, useCallback, useRef, useState } from "react";
+
+import { v4 } from "uuid";
+import useMediaQuery from "@Utils/hooks/useMediaQuery";
+
+const KanbanCard = ({ provided, taskIndex, task, deleteTask, updateTaskName }) => {
+  const [cardEditMode, setCardEditMode] = useState(false);
+  const [cardInputValue, setCardInputValue] = useState(task.name);
+
+  const enableTaskEdit = () => {
+    setCardInputValue(task.name);
+    setCardEditMode(true);
+  }
```

Comment:
We should add a rule here that if it is in edit mode, we should not bet able to move it